spec-0012/index.md

```diff
@@ -0,0 +1,185 @@
+     containing more than one `**` operator. This code would be executed as `x**(y**z)`
+     following the implicit order, but the explicit parentheses should be included for
+     clarity.
+   - In the expression `t**v*x**y + z`, no spaces are used around the multiplication
+     operator due to the presence of the lower-priority addition operator. However,
+     this would lead to `t**v*x**y` being an implicit subexpression without spaces
+     containing more than one `**` operator. This code would be executed as 
+    `(t**v)*(x**y) + z`, but the explicit parentheses should be included for clarity.
+   - In the expression `z + x**y/w`, no spaces are used around the division operator
+     due to the presence of the lower-priority addition operator. However, this would
+     lead to `x**y/w` being an implicit subexpression without spaces containing `**`
+     to the left of another operator. Options for refactoring include the addition of
+     parentheses (e.g. `z + (x**y)/w`) or pre-multiplying the exponential by a
+     fraction (i.e. `x + 1/w*x**y`).
+6. Simplify combinations of unary and binary `+` and `-` operators when possible.
+   For example,
+   - prefer `x + y` over `x + +y`,
+   - prefer `x + y` over `x - -y`,
+   - prefer `x - y` over `x - +y`, and
+   - prefer `x - y` over `x + -y`.
+7. If required to satisfy other style requirements, include line breaks before
+   the outermost explicit subexpression possible. For example, if
+   `t + (w + (x + (y + z))))` must be broken, prefer
+   ```python3
+   (t 
+    + (w + (x + (y + z)))))
+   ```
+   over
+   ```python3
+   (t + (w + (x + (y 
+                   + z)))))
+   ```
+   If there are multiple candidates, include the break at the first opportunity.
+8. If line breaks must occur within a compound subexpression, the break should
+   be placed before the operator with lowest priority. For example, if
+   (x + y*z) must be broken, prefer
+   ```python3
+   (x
+    + y*z)
+   ```
+   over
+   ```python3
+   (x + y
+    * z)
+   ```
+   If there are multiple candidates, include the break at the first opportunity.
+9. Any of the preceeding rules may be broken if there is a clear reason to do so.
+    - *Conflict with other style rules*. For example, there is not supposed to be
+      whitepace surrounding the `**` operator, but one can imagine a chain of `**`
+      operations that exhausts the character limit of a line.
+    - *Domain knowledge*. For instance, in the expression
+      `t = (x + y) - z`, it may be important to emphasize that the addition should be
+      performed first for numerical reasons or because `(x + y)` is a conceptually
+      important quantity. In such cases, consider adding a comment, e.g.
+      ```python3
+      t = (x + y) - z  # perform `x + y` first for precision
+      ```
+      or breaking the expressions into separate logical lines, e.g.
+      ```python3
+      w = x + y
+      t = w - z
+      ```
```

Comment:
Just to be explicit: is the intention that autoformatters (like Ruff) would support automatically formatting per this spec?

Reply:
I think this would be hard or even impossible because the formatter has no domain knowledge and requires explicit rules on how to deal with conflicts